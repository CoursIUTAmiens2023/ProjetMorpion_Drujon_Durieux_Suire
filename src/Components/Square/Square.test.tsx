import { describe, expect, it } from "@jest/globals"
import { render } from "@testing-library/react"
import { Square } from "."

describe("Square", () => {
    it("should match the snapshot when value null", () => {
        const { container } = render(<Square value={null} onClick={() => {}} />)
        expect(container).toMatchSnapshot()
    })
    it("should match the snapshot when value X", () => {
        const { container } = render(<Square value="X" onClick={() => {}} />)
        expect(container).toMatchSnapshot()
    })
    it("should match the snapshot when value O", () => {
        const { container } = render(<Square value="O" onClick={() => {}} />)
        expect(container).toMatchSnapshot()
    })
})
