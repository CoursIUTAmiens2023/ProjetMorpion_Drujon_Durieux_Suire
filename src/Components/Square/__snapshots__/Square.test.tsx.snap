// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square should match the snapshot when value O 1`] = `
<div>
  <button
    class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-blue-600"
    id="0"
  >
    <span
      class="text-4xl"
    >
      O
    </span>
  </button>
</div>
`;

exports[`Square should match the snapshot when value X 1`] = `
<div>
  <button
    class="size-32 border-2 border-[#364439] bg-[#f1e8f7] text-red-600"
    id="0"
  >
    <span
      class="text-4xl"
    >
      X
    </span>
  </button>
</div>
`;

exports[`Square should match the snapshot when value null 1`] = `
<div>
  <button
    class="size-32 border-2 border-[#364439] bg-[#f1e8f7] "
    id="0"
  >
    <span
      class="text-4xl"
    />
  </button>
</div>
`;
